extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryChangeLogSource.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.rya.streams.querymanager;
+
+import java.util.Set;
+
+import org.apache.rya.streams.api.queries.QueryChange;
+import org.apache.rya.streams.api.queries.QueryChangeLog;
+
+import com.google.common.util.concurrent.Service;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A source service for {@link QueryChangeLog}s that periodically checks
+ * for new {@link QueryChangeLog}s.
+ *
+ * When {@link QueryChange}s are found, the {@link SourceListener}s are
+ * notified with the {@link QueryChangeLog}s
+ */
+@DefaultAnnotation(NonNull.class)
+public interface QueryChangeLogSource extends Service {
+
+    /**
+     * Adds a {@link SourceListener} to be notified when {@link QueryChangeLog}s
```

Comment:
Maybe indicate it will be notified with all that already exist upon registration.

Reply:
done